src/bsoncxx/types.hpp

```diff
@@ -346,10 +346,6 @@ BSONCXX_INLINE bool operator==(const b_date& lhs, const b_date& rhs) {
 ///
 /// A BSON null value.
 ///
-/// @warning
-///   This BSON type is used internally by the MongoDB server - use by clients
-///   is discouraged.
-///
```

Comment:
I believe that the intention was that the warning only apply to the deprecated types: undefined, dbpointer, and symbol.

I think the best thing to do is to @ deprecate them with doxygen in the comments, and then officially deprecate the types with BSONCXX_DEPRECATED tags on the types.

This might create some problems when we tackle https://jira.mongodb.org/browse/CXX-936, since then we will trip on our own deprecations.

But I think we should tackle that later, rather than sooner.